src/wp-includes/class-wp-theme.php

```diff
@@ -1250,9 +1250,17 @@ public function get_post_templates() {
 			}
 
 			if ( current_theme_supports( 'block-templates' ) ) {
+				$block_templates = get_block_templates( array(), 'wp_template' );
 				foreach ( get_post_types( array( 'public' => true ) ) as $type ) {
-					$block_templates = get_block_templates( array( 'post_type' => $type ), 'wp_template' );
 					foreach ( $block_templates as $block_template ) {
+						if ( ! $block_template->is_custom ) {
```

Comment:
I feel maybe this should be internal to `get_block_templates` (an argument of the query) but I'm fine with this approach as well.